cmd/podman/containers/exec.go

```diff
@@ -161,3 +177,42 @@ func exec(_ *cobra.Command, args []string) error {
 	fmt.Println(id)
 	return nil
 }
+
+func execWait(ctr string, seconds int32) error {
+	maxDuration := time.Duration(seconds) * time.Second
+	interval := 100 * time.Millisecond
+
+	cond, err := define.StringToContainerStatus("running")
+	if err != nil {
+		return err
+	}
+	waitOptions.Condition = append(waitOptions.Condition, cond)
+
+	startTime := time.Now()
+	for time.Since(startTime) < maxDuration {
+		inspectData, _, err := registry.ContainerEngine().ContainerInspect(context.Background(), []string{ctr}, entities.InspectOptions{})
+		if err != nil {
+			return err
+		}
+		if len(inspectData) == 1 && inspectData[0].State.Status != "running" {
+			continue
+		}
```

Comment:
Inspect is quite heavy and expensive. Is this only to prevent the case where ContainerWait() would block forever when the container is created but not running?
Right now there is no point to even call ContainerWait() in this case.

If this is the case the correct solution is to pass a context with a timeout to ContainerWait(), create this at the top of the function then just pass this context to ContainerWait()
```
ctx, cancel := context.WithTimeout(registry.Context(), 5*time.Second)
defer cancel()
```
ContainerWait() should then return a ErrCanceled error if we hit the context timeout.

Reply:
`ErrCanceled` says `cancelled by user`. Would that be the right error for timeouts?

Reply:
so updating it this way required me to update the tests a little as well. PTAL. Also, I'd rather leave friendlier error messages for a followup PR to catch all cases. Hoping we can get this merged ( assuming CI is green).